### PR TITLE
Specific vote error message if you don't have enough karma

### PR DESCRIPTION
**Pull request by darkruby501**
Current situation in image. Change message to be more specific about vote karma if that's the reason.

image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204690577737351) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/voteServer.ts [hunk 1]
```diff
@@ -223,6 +224,8 @@ export const performVoteServer = async ({ documentId, document, voteType, extend
   const collectionVoteType = `${collectionName.toLowerCase()}.${voteType}`
 
   if (!user) throw new Error("Error casting vote: Not logged in.");
+  
+  if (isLW && user.karma < lowKarmaUserVotingCutoffKarmaSetting.get()) throw new Error(`Error casting vote: You must have ${lowKarmaUserVotingCutoffKarmaSetting.get()}+ karma to vote.`);
```
Comment (b0b3rt) [inline packages/lesswrong/server/voteServer.ts:228]: You don't need to add this check, this is already checked in `userCanVote` (on the line below).  In addition to returning `fail` it also returns a `reason: string` if `fail: true`, so you can throw whatever reason it returns as the error.

Comment (darkruby501) [reply packages/lesswrong/server/voteServer.ts:228]: Ah, good point. I originally had the throw error in there (not using the returned reason), but then though mistakenly it was cleaner to put it above

File: packages/lesswrong/server/voteServer.ts [hunk 2]
```diff
@@ -223,11 +223,11 @@ export const performVoteServer = async ({ documentId, document, voteType, extend
   const collectionVoteType = `${collectionName.toLowerCase()}.${voteType}`
 
   if (!user) throw new Error("Error casting vote: Not logged in.");
-
+
```
Comment (b0b3rt) [inline packages/lesswrong/server/voteServer.ts:226]: nit: remove extra newline

**Review by b0b3rt — APPROVED** [review]
lgtm, with a nit

File: packages/lesswrong/server/voteServer.ts (hunk 2 above)
Comment (darkruby501) [reply packages/lesswrong/server/voteServer.ts:226]: I'm confused, there's no extra space compared to Master

**Comment by jimrandomh** [comment]
Note that there's still a mysterious bug here which this doesn't address, which is why the vote buttons are disabled for insufficient-karma users on Chrome, but not disabled on Firefox. If they were disabled correctly on Firefox, then this code path would never be exercised (except in the very rare case where you lost karma in between when the page loaded and when you clicked the vote button).

**Comment by darkruby501** [comment]
Yeah, I know, but I wanted to a quick fix out without hunting that down.

**Review by jpaddison3 — APPROVED** [review]

**Pull request merged**